src/components/timepicker/timepicker.ts

```diff
@@ -211,33 +233,80 @@ export class MTimepicker extends ModulVue {
         }
     }
 
-    private positionScroll(el: Element): void {
-        el.scrollTop = Math.round(el.scrollTop / 44) * 44;
+    private onOk(): void {
+        if (this.internalHour && this.internalMinute) {
+            this.emitTime();
+            this.open = false;
+        }
     }
 
-    private selectHour(hour: number): void {
-        this.tempHour = hour;
+    private onClose(): void {
+        if (!this.internalHour || !this.internalMinute) {
+            this.resetPopupTime();
+        }
     }
 
-    private selectMinute(minute: number): void {
-        this.tempMinute = minute;
-    }
+    ///////////////////////////////////////
 
-    private onOk(): void {
-        this.selectedHour = this.tempHour;
-        this.selectedMinute = this.tempMinute;
-        this.emitChange(this.selectedHour, this.selectedMinute);
-        this.open = false;
+    public get currentTime(): string {
+        return this.internalTime;
     }
 
-    private emitChange(hour: number, minute: number): void {
-        if (this.duration) {
-            this.$emit('change', moment.duration(hour + ':' + minute));
+    public set currentTime(value: string) {
```

Comment:
On pourrait aussi faire une variable currentTime et faire un watch dessus et ensuite effectuer les modification du model interne dans la fonction du watch

Personnellement je trouve ça plus simple a comprendre que mettre de la logique dans un setter.

Reply:
source: https://vuejs.org/v2/guide/computed.html

That’s why for any complex logic, you should use a computed property.

When you have some data that needs to change based on some other data, it is tempting to overuse watch - especially if you are coming from an AngularJS background. However, it is often a better idea to use a computed property rather than an imperative watch callback. Consider this example: